graphium/finetuning/finetuning.py

```diff
@@ -0,0 +1,56 @@
+from typing import Iterable, List, Dict, Tuple, Union, Callable, Any, Optional, Type
+
+import pytorch_lightning as pl
+
+from torch.optim.optimizer import Optimizer
+from pytorch_lightning.callbacks import BaseFinetuning
+
+
+class MolecularFinetuning(BaseFinetuning):
+    def __init__(
+        self,
+        cfg,
+        train_bn: bool = False
+
+    ):
+        """
+        Finetuning logic
+        """
+        super().__init__()
+
+        cfg_finetune = cfg['trainer']['finetuning']
+        self.depth = cfg_finetune['added_layers']['depth']
+        try:
+            self.depth += cfg_finetune['add_finetune_depth']
+        except:
+            pass
```

Comment:
I guess this is in case `'add_finetune_depth'` is missing? You an do this instead, which puts a default value of 0 if the key is missing. But I would suggest changing the `__init__` altogether
```suggestion
        self.depth += cfg_finetune.get("add_finetune_depth", 0)
```